extensions/2.0/Khronos/KHR_texture_basisu/README.md

```diff
@@ -0,0 +1,173 @@
+                "KHR_texture_basisu": {
+                    "source": 1
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "mimeType": "image/png",
+            "bufferView": 1
+        },
+        {
+            "mimeType": "image/ktx2",
+            "bufferView": 2
+        }
+    ]
+}
+```
+
+### Using Without a Fallback
+
+To use KTX v2 image with Basis Universal supercompression without a fallback, define `KHR_texture_basisu` in both `extensionsUsed` and `extensionsRequired`. The `texture` object will then have its `source` property omitted as shown below.
+
+```json
+{
+    "asset": {
+        "version": "2.0"
+    },
+    "extensionsUsed": [
+        "KHR_texture_basisu"
+    ],
+    "extensionsRequired": [
+        "KHR_texture_basisu"
+    ],
+    "textures": [
+        {
+            "extensions": {
+                "KHR_texture_basisu": {
+                    "source": 0
+                }
+            }
+        }
+    ],
+    "images": [
+        {
+            "uri": "image.ktx2"
+        }
+    ]
+}
+```
+
+### JSON Schema
+
+[texture.KHR_texture_basisu.schema.json](schema/texture.KHR_texture_basisu.schema.json)
+
+## KTX v2 Images with Basis Universal Supercompression
+
+To cover a broad range of use cases, this extension allows two Basis Universal formats: ETC1S with BasisLZ and UASTC with optional Zstandard compression.
+
+- KTX header fields for ETC1S with BasisLZ payloads
+  - `supercompressionScheme` must be `1` (BasisLZ).
+  - DFD `colorModel` must be `KHR_DF_MODEL_ETC1S`.
+  - RGB textures must use a single DFD channel with value `KHR_DF_CHANNEL_ETC1S_RGB`.
+  - RGBA textures must use two DFD channels with values `KHR_DF_CHANNEL_ETC1S_RGB` and `KHR_DF_CHANNEL_ETC1S_AAA`.
+
+- KTX header fields for UASTC payloads
+  - `supercompressionScheme` must be `0` (None) or `2` (Zstandard).
+  - DFD `colorModel` must be `KHR_DF_MODEL_UASTC`.
+  - RGB textures must use a single DFD channel with value `KHR_DF_CHANNEL_UASTC_RGB`.
+  - RGBA textures must use a single DFD channel with value `KHR_DF_CHANNEL_UASTC_RGBA`.
+
+Regardless of the format used, these additional restrictions apply for compatibility reasons:
+
+- Swizzling metadata (`KTXswizzle`) must be `rgba` or omitted.
+- Orientation metadata (`KTXorientation`) must be `rd` or omitted.
+- Colorspace information in the DFD must match the expected usage, namely:
+  - For textures with color data, `colorPrimaries` must be set to `KHR_DF_PRIMARIES_BT709`; `transferFunction` to `KHR_DF_TRANSFER_SRGB`.
+  - For textures with non-color data, `colorPrimaries` must be set to `KHR_DF_PRIMARIES_UNSPECIFIED`; `transferFunction` to `KHR_DF_TRANSFER_LINEAR`.
+- `KHR_DF_FLAG_ALPHA_PREMULTIPLIED` flag must not be set unless the material's specification requires premultiplied alpha.
+- When a texture refers to a sampler with mipmap minification or when the sampler is undefined, the KTX image should contain a full mip pyramid.
```

Comment:
Consider using uppercase for "MUST" and "SHOULD" throughout — I'm assuming you intend their [normative meanings](https://www.ietf.org/rfc/rfc2119.txt), but that would be easy for readers to miss.

Reply:
I don't think the main spec uses that style although it would be great to adopt it. Do you think it's fine to start with this extension?

Reply:
Most appearances of "should" in the main spec are in implementation notes (based on a quick scan anyway) and so are fairly clear even if the reader isn't looking for normative language. This bulleted list would be easier to misinterpret, I think, and needs either:

(a) "Implementation Note" language on optional items, or  
(b) "SHOULD" language for optional items.

I think I'd vote for (b), even though we haven't done so in the main spec. This could be changed later, but more likely, we may even want to adopt it in the main spec.